File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] arr) {
        HashSet<Integer> set = new HashSet<>();
        for(int num: arr){
            set.add(num);
        }

        int streak = 0;
        for(int num: set){
            if(!set.contains(num-1)){
                int currentStreak = 1;
                int currentNum = num;
                while(set.contains(currentNum+1)){
                    currentStreak++;
                    currentNum++;
                }
                streak = Math.max(streak,currentStreak);
            }
        }
        return streak;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if(len(nums) == 0):
            return 0
        nums.sort()
        ans = 0
        cnt = 1
        for i in range(1,len(nums)):
            if(nums[i] == nums[i-1]):
                continue
            if(nums[i] == nums[i-1]+1):
                cnt+=1
            else:
                ans = max(ans,cnt)
                cnt= 1
        return max(ans,cnt)
